appcelerator/titanium_mobile
# fix(android): incremental build creates duplicate "bootstrap.json" entries

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27904

**Summary:**
This is a regression as of Titanium 8.1.0. Every incremental build duplicates a bootstrap entry in the "bootstrap.json" file, causing its `execute()` function to be wrongly invoked multiple times on app startup.

**Test:**
1. Create a Titanium app project.
2. Copy below `test.bootstrap.js` to `Resources` directory. Do not change the file name.
3. Build and run on Android.
4. Notice a bootstrap dialog appears once. _(This is good.)_
5. Build and run on Android **again**.
6. Verify that bootstrap dialog only appears once.

test.bootstrap.js
```javascript
Ti.API.info("### Bootstrap was required-in.");

var wasExecuted = false;
exports.execute = function(finished) {
	Ti.API.info("### Bootstrap execute() method was called.");
	var message = "This is the bootstrap dialog.";
	if (wasExecuted) {
		message = "Uh-oh!!! Bootstrap was wrongly executed again.";
	}
	var dialog = Ti.UI.createAlertDialog({
		message: message,
		buttonNames: ['OK'],
		cancel: 0,
		persistent: true,
	});
	dialog.addEventListener('click', finished);
	dialog.addEventListener('cancel', finished);
	dialog.show();
	wasExecuted = true;
};
```
